src/operations/blas3/gemm_no_local_full_vec.hpp

```diff
@@ -381,7 +381,7 @@ class Gemm<input_t, output_t, DoubleBuffer, NbcA, NbcB, ClSize, tile_type,
       auto C = orig_C;
 
       /* register array used to store the result*/
-      value_t reg_res[item_rows * item_cols];
+      element_t reg_res[item_rows * item_cols];
```

Comment:
Can you explain whats happening here?

Reply:
Thank you @hjabird for the question, I'll do my best.
`value_t` is set to be the same type of the input, see `input_t`, removing the `const` qualifier. Then this type is used also for the output, it works as long as the input and the output are of the *same* type. With mixed `gemm`, it doesn't work anymore and it breaks at compile time.
I did not personally come up with this change/work around, but I am mimicking how it is implemented in other gemm implementations. The idea is that the scalar that multiplies the matrices, it is of the correct output type and, as far as I can see, it was selected and used also to store intermediate results for the output.